### PR TITLE
ldexp extension

### PR DESCRIPTION
**Pull request by pillarxyz**

### CODE REVIEW

File: ivy/functional/backends/numpy/experimental/elementwise.py [hunk 1]
```diff
@@ -429,3 +429,13 @@ def xlogy(
 
 def real(x: np.ndarray, /, *, out: Optional[np.ndarray] = None) -> np.ndarray:
     return np.real(x)
+
+
+def ldexp(
+    x: np.ndarray,
+    i: Union[np.ndarray, int, list, tuple],
```
Comment (nassimberrada) [inline ivy/functional/backends/numpy/experimental/elementwise.py:436]: It doesn't change much really but only torch uses the notation `x` and `i` for the inputs, might be better to stick with `x1` and `x2` as is the case numpy and jax ? I see you've used the latter notation for the testing when you passed `x1=x[0]` and `x2=x[1]` , you've also used that in the jax backend. In general you want to keep consistency in input format across the backends. It's at the front-end level that we mimic the patterns of each framework separately

File: ivy/functional/ivy/experimental/elementwise.py [hunk 2]
```diff
@@ -1358,3 +1358,40 @@ def binarizer(
...
+@to_native_arrays_and_back
+@handle_out_argument
+@handle_nestable
+@handle_exceptions
+@handle_array_like_without_promotion
+def ldexp(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    i: Union[ivy.Array, ivy.NativeArray],
```
Comment (nassimberrada) [inline ivy/functional/ivy/experimental/elementwise.py:1372]: I believe the second array input (whether you call it `i` or `x2`), should be positional-only and placed before the `/,` sign

File: ivy/container/elementwise.py [hunk 3]
```diff
@@ -8846,3 +8846,94 @@ def trapz(
...
+        Returns
+        -------
+        ret
+            container including x * 2**i.
+
+        Examples
+        --------
+        With one :class:`ivy.Container` input:
+        >>> x = ivy.Container(a=ivy.array([1, 2, 3]), b=ivy.array([1, 5, 10]))
+        >>> i = ivy.Container(a=ivy.array([1, 2, 3]), b=ivy.array([1, 5, 10]))
```
Comment (nassimberrada) [inline ivy/container/elementwise.py:8932]: You've already made changes to the experimental version of the containers so this should be removed

**Review by nassimberrada — CHANGES_REQUESTED** [review]
Hey @pillarxyz, I've looked through your PR and it's looking mostly good ! I left a couple comments maybe you could address those and make appropriate modifications across the files ? Let me know when you feel it's good to go and the tests are passing, I'll give it another review then. Thanks !

File: ivy/array/experimental/elementwise.py [hunk 4]
```diff
@@ -1164,3 +1164,37 @@ def binarizer(
             Binarized output data
         """
         return ivy.binarizer(self._data, threshold=threshold, out=out)
+
+    def ldexp(
+        self: ivy.Array,
+        /,
```
Comment (nassimberrada) [inline ivy/array/experimental/elementwise.py:1170]: I think the second array input is missing here, and in the docstring as well

File: ivy/container/experimental/elementwise.py [hunk 5]
```diff
@@ -3225,3 +3225,94 @@ def binarizer(
             map_sequences=map_sequences,
             out=out,
         )
+
+    @staticmethod
+    def static_ldexp(
+        x: Union[ivy.Array, ivy.NativeArray, ivy.Container],
+        i: Union[ivy.Array, ivy.NativeArray, ivy.Container],
```
Comment (nassimberrada) [inline ivy/container/experimental/elementwise.py:3232]: I see you've changed Numpy's arguments to `x1` and `x2` , could you apply those changes here as well ?

Comment (nassimberrada) [inline ivy/container/experimental/elementwise.py:3232]: I see you've changed Numpy's arguments to `x1` and `x2` , could you apply those changes here as well ?

File: ivy/container/experimental/elementwise.py [hunk 6]
```diff
@@ -3225,3 +3225,94 @@ def binarizer(
...
+        }
+        """
+        return ContainerBase.cont_multi_map_in_function(
+            "ldexp",
+            x,
+            key_chains=key_chains,
+            to_apply=to_apply,
+            prune_unapplied=prune_unapplied,
+            map_sequences=map_sequences,
+            i=i,
```
Comment (nassimberrada) [inline ivy/container/experimental/elementwise.py:3278]: Also the second array input is not a keyword argument as specified so no need to write is as argument = argument

File: ivy/container/experimental/elementwise.py [hunk 7]
```diff
@@ -3225,3 +3225,94 @@ def binarizer(
...
+            out=out,
+        )
+
+    def ldexp(
+        self: ivy.Container,
+        i: Union[ivy.Array, ivy.NativeArray, ivy.Container],
```
Comment (nassimberrada) [inline ivy/container/experimental/elementwise.py:3284]: Likewise should be changed to `x1` and `x2` alongside the docstrings

File: ivy/functional/backends/tensorflow/experimental/elementwise.py [hunk 8]
```diff
@@ -613,3 +613,12 @@ def real(
     out: Optional[Union[tf.Tensor, tf.Variable]] = None,
 ) -> Union[tf.Tensor, tf.Variable]:
     return tf.math.real(x)
+
+def ldexp(
+    x: Union[tf.Tensor, tf.Variable],
+    i: Union[tf.Tensor, tf.Variable],
```
Comment (nassimberrada) [inline ivy/functional/backends/tensorflow/experimental/elementwise.py:619]: Likewise should be changed to `x1` and `x2` alongside the function calls

File: ivy/functional/backends/torch/experimental/elementwise.py [hunk 9]
```diff
@@ -433,3 +433,13 @@ def xlogy(
 
 def real(x: torch.Tensor, /, *, out: Optional[torch.Tensor] = None) -> torch.Tensor:
     return torch.real(x)
+
+
+def ldexp(
+    x: torch.Tensor,
+    i: Union[int, torch.Tensor],
```
Comment (nassimberrada) [inline ivy/functional/backends/torch/experimental/elementwise.py:440]: Likewise should be changed to `x1` and `x2` alongside the function calls

File: ivy/functional/ivy/experimental/elementwise.py [hunk 10]
```diff
@@ -1358,3 +1358,40 @@ def binarizer(
...
+@handle_array_like_without_promotion
+def ldexp(
+    x1: Union[ivy.Array, ivy.NativeArray],
+    x2: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """
+    Returns x * (2**i), element-wise.
```
Comment (nassimberrada) [inline ivy/functional/ivy/experimental/elementwise.py:1376]: Likewise the docstrings should reflect  `x1` and `x2`

**Review by nassimberrada — CHANGES_REQUESTED** [review]
@pillarxyz haven't taken a look at the tests yet but I've left a couple other comments that should make things almost ready once adjusted to, could you give those a look ? Thanks !

**Review by nassimberrada — CHANGES_REQUESTED** [review]
@pillarxyz haven't taken a look at the tests yet but I've left a couple other comments that should make things almost ready once adjusted to, could you give those a look ? Thanks !

**Comment by pillarxyz** [comment]
@VedPatwardhan what happened with the PR ?

**Comment by pillarxyz** [comment]
#11113

**Pull request closed without merging**